api/src/main/java/com/datastrato/gravitino/authorization/Group.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.authorization;
+
+import com.datastrato.gravitino.Auditable;
+import com.datastrato.gravitino.annotation.Evolving;
+import java.util.List;
+
+/** The interface of a Group. The Group is the entity which contains users. */
+@Evolving
+public interface Group extends Auditable {
```

Comment:
How does Gravitino know which user belongs to which group?

Reply:
We can get the groups by using user name in the Authenticator. We use user name to retrieve the user from the external user system like LDAP, RANGER, IAM.

I will add `GroupMappingServiceProvider` for accomplish this feature.
```
interface GroupMappingServiceProvider {
   List<String> getGroups(String user);
}
```